src/mcx_core.cu

```diff
@@ -1462,6 +1462,29 @@ __device__ inline int launchnewphoton(MCXpos* p, MCXdir* v, Stokes* s, MCXtime*
                         rotatevector(v, 1.f, 0.f, sphi, cphi);
                     }
 
+                    if (MCX_SRC_SLIT && (launchsrc->param2.x > 0.f || launchsrc->param2.y > 0.f)) {
```

Comment:
the first condition should be `gcfg->srctype == MCX_SRC_SLIT`. I will post my fix, just leave here for a note.